grisp/default/common/build/files/erts/emulator/sys/unix/erl_main.c

```diff
@@ -211,6 +247,44 @@ void fatal_atexit(void) {
   }
 }
 
+char* silence_erl_console(char *args) {
+  char *pos, *new_args;
+  size_t added_length, new_len, length_before;
+
+  /* Check if the arguments are for the erlang VM */
+  if (strncmp(args, "erl.rtems", strlen("erl.rtems")) != 0) {
+      // If not, return the original string
+      return args;
+  }
+
+  /* Find the position of the "--" */
+  pos = strstr(args, "--");
+  if (!pos) {
+    /* If "--" is not found, return the original string */
+    return args;
+  }
+
+  /* Calculate the new string length */
+  added_length = strlen(" -noshell -noinput");
+  new_len = strlen(args) + added_length;
+  new_args = (char*) malloc(new_len + 1); /* +1 for the null terminator */
+  if (!new_args) {
+    /* Memory allocation failed, return the original string */
+    return args;
+  }
+
+  /* Copy the parts of the original string and insert the new strings */
+  length_before = pos - args + 2; /* +2 to include the "--" */
+  strncpy(new_args, args, length_before);
+  strcpy(new_args + length_before, " -noshell -noinput");
+  strcpy(new_args + length_before + added_length, pos + 2);
+
+  /* Release the original string */
+  free(args);
```

Comment:
This assumes args was allocated.  To free it in a function is not the cleanest way of handling that.

Reply:
mmmm, ok, I can change that

Reply:
Freeing the memory here make it a lot more simple when the caller do not have to worry about if the return pointer is the original value or some new modified value. To handle this without freeing the memory here it would have to return a code, and gat a pointer reference for the result, so the caller can decide to free or not the original pointer.